packages/contracts-core/contracts/manager/AgentManager.sol

```diff
@@ -98,9 +98,12 @@ abstract contract AgentManager is MessagingBase, AgentManagerEvents, IAgentManag
         if (_agentDispute[notaryIndex].flag != DisputeFlag.None) revert NotaryInDispute();
         _disputes.push(OpenedDispute(guardIndex, notaryIndex, 0));
         // Dispute is stored at length - 1, but we store the index + 1 to distinguish from "not in dispute".
-        uint256 disputePtr = _disputes.length;
-        _agentDispute[guardIndex] = AgentDispute(DisputeFlag.Pending, uint88(disputePtr), address(0));
-        _agentDispute[notaryIndex] = AgentDispute(DisputeFlag.Pending, uint88(disputePtr), address(0));
+        // TODO: check if we really need to use 88 bits for dispute indexes. Every dispute ends up in one of
+        // the agents being slashed, so the number of disputes is limited by the number of agents (currently 2**32).
+        // Thus we can do the unsafe cast to uint88.
+        uint88 disputePtr = uint88(_disputes.length);
```

Comment:
does this comment logic apply if at any point number of agents is changed (is this possible)?

Reply:
The overall logic stands: the amount of disputes is always limited by the max total amount of agents (every dispute removes an agent).

If the Agent Merkle tree is expanded, and more than 2**32 historical agents could exist, it's very likely that we would still go for something lower than 88 bits there, as adding of the new agents would be naturally limited by providing the bonds.